### DISCUSSION TITLE
Devtoys - Data Processing Location

### DISCUSSION THREAD

**Discussion by Barnoux** [DevToys-app/DevToys#1336] [Q&A] [answered]
Hello 😄 

I didn't find documentation on this subject, but i maybe miss something.

Where is the data processed when using the app? Is it processed locally on the device or externally on remote servers?

Regards,

BBA

**Comment by veler** [comment, marked as answer]
Hello,
This is explained in a popup the first time you start devtoys on your machine : 

All the default tools of DevToys work completely offline and everything is treated pouvant on your machine. Nothing goes to the internet.

However, we have no control or guarantee on what third party tools do (installed through extensions). If you care about safety, the best is to not install this party extensions, or to audit them before using.

**Reply by Barnoux** [reply]
Ok Thank you at least this subject is now aborded in QA 😄
